testsuite/standalone/src/test/java/org/wildfly/core/test/standalone/mgmt/api/core/JdkModuleDependencyTestCase.java

```diff
@@ -0,0 +1,220 @@
+import org.jboss.as.test.integration.management.ManagementOperations;
+import org.jboss.as.test.shared.PermissionUtils;
+import org.jboss.dmr.ModelNode;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.exporter.ZipExporter;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.Assert;
+import org.junit.Assume;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.wildfly.core.testrunner.ManagementClient;
+import org.wildfly.core.testrunner.WildflyTestRunner;
+
+import javax.inject.Inject;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Properties;
+import java.util.PropertyPermission;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Test case checking whether a module can depend on standard OpenJDK 9+ platform modules (or their emulation in case of Java 8).
+ *
+ * @author <a href="mailto:mjurc@redhat.com">Michal Jurc</a> (c) 2018 Red Hat, Inc.
+ */
+@RunWith(WildflyTestRunner.class)
+public class JdkModuleDependencyTestCase {
+
+    public static final String DEPLOYMENT_NAME_SUFFIX = "-test-dep.jar";
+
+    /**
+     * A set of modules that we need to emulate to be compatible with JDK8.
+     */
+    public static final Set<String> REQUIRED_1_8_EMULATED_MODULES = Stream.of("java.base", "java.compiler", "java.datatransfer",
+            "java.desktop", "java.instrument", "java.jnlp", "java.logging", "java.management", "java.management.rmi",
+            "java.naming", "java.prefs", "java.rmi", "java.scripting", "java.security.jgss", "java.security.sasl",
+            "java.smartcardio", "java.sql", "java.sql.rowset", "java.xml", "java.xml.crypto", "javafx.base", "javafx.controls",
+            "javafx.fxml", "javafx.graphics", "javafx.media", "javafx.swing", "javafx.web", "jdk.accessibility", "jdk.attach",
+            "jdk.compiler", "jdk.httpserver", "jdk.jartool", "jdk.javadoc", "jdk.jconsole", "jdk.jdi", "jdk.jfr", "jdk.jsobject",
+            "jdk.management", "jdk.management.cmm", "jdk.management.jfr", "jdk.management.resource", "jdk.net", "jdk.plugin.dom",
+            "jdk.scripting.nashorn", "jdk.sctp", "jdk.security.auth", "jdk.security.jgss", "jdk.unsupported", "jdk.xml.dom",
+            "org.jboss.modules").collect(Collectors.toCollection(HashSet::new));
+
+    /**
+     * A set of all non-internal, non-deprecated JDK9 API modules.
+     *
+     * The jdk.* modules are omitted as they are not guaranteed to be present on any given JDK. The same applies for javafx.*
+     * modules.
+     * The modules that are deprecated and marked for removal with JDK9 are omitted as well.
+     *
+     * @see <a href="https://docs.oracle.com/javase/9/docs/api/overview-summary.html">Overview (Java SE 9 &amp; JDK 9 )</a>
+     */
+    public static final Set<String> REQUIRED_9_MODULES = Stream.of("java.base", "java.compiler", "java.datatransfer",
```

Comment:
@dmlloyd I've removed the ``jdk.*``, ``javafx`` and ``@Deprecated`` modules from this list. There is one last failure remaining - ``java.jnlp``.

Reply:
OK.  I did not bother to include `java.jnlp` because it was removed in Java 11 though.  Do you think we should include it anyway?

Reply:
I don't think there's much point to it now - I will remove it from the sets.